x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/FoldNull.java

```diff
@@ -41,8 +41,9 @@ public Expression rule(Expression e) {
             if (Expressions.isGuaranteedNull(in.value())) {
                 return Literal.of(in, null);
             }
-        } else if (e instanceof Alias == false
-            && e.nullable() == Nullability.TRUE
+        } else if (e instanceof Alias == false && e.nullable() == Nullability.TRUE
+        // Categorize function stays as a STATS grouping (It isn't moved to an early EVAL like other groupings),
```

Comment:
The story of null folding is not complete (PropagateEvalFoldables has missing pieces), but I appreciate adding this comment here. [I was kind of confused and surprised to see here this out-of-nowhere `instanceof Categorize`](https://github.com/elastic/elasticsearch/pull/118176).